recipes/sherloc/inference/sherloc_utils/context_manager.py

```
@@ -0,0 +1,629 @@
+
+        Args:
+            turns: Current conversation turns
+            loop_info: Information about the detected loop
+
+        Returns:
+            Modified turns with intervention message
+        """
+        # Parse the repeated call to provide specific guidance
+        try:
+            repeated_call = json.loads(loop_info["repeated_call"])
+            tool_name = list(repeated_call.keys())[0]
```

Ruff (0.16.2)

[warning] Prefer `next(iter(repeated_call.keys()))` over single element slice

Replace with `next(iter(repeated_call.keys()))`

(RUF015)